students/kimjy/westagram/users/views.py

```diff
@@ -1,3 +1,47 @@
-from django.shortcuts import render
+import json
+import re
 
+from django.views      import View
+from django.http       import JsonResponse, Http404
+from django.db.utils   import IntegrityError
+from users.models      import User
+from users.check_items import (check_character_policy, 
+                               check_dictionary_keys,
+                               login_key_list,
+                               password_policy,
+                               email_policy)
+
+class SignUpView(View):
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+
+            check_dictionary_keys(data, login_key_list)
+
+            username = data['username']
+            email    = data['email'] 
+            password = data['password']
+            
+            check_character_policy(password_policy, password)
+            check_character_policy(email_policy, email)
+
+            if User.objects.filter(email_address=email).exists():
+                raise IntegrityError("EMAIL_DUPLICATED_ERROR")
+            else:
+                user = User.objects.create(
+                            username      = username,
+                            password      = password,
+                            email_address = email,
+                )
+                if 'phone_number' in data:
+                    user.phone_number = data['phone_number']
+                    user.save()
```

Comment:
1. if를 통과하게 되면 error가 raise되기 때문에 else를 굳이 안 사용해도 좋을 것 같습니다.
2. dictionary의 get method을 사용하면 key가 없을 때 default 값을 정할 수 있습니다. 그렇게 된다면 if 구문을 줄일 수 있습니다. 한 번 고민해 보세요.

```suggestion
            if User.objects.filter(email_address=email).exists():
                raise IntegrityError("EMAIL_DUPLICATED_ERROR")

            user = User.objects.create(
                        username      = username,
                        password      = password,
                        email_address = email,
            )
            
            if 'phone_number' in data:
                user.phone_number = data['phone_number']
                user.save()
```

Reply:
말씀해주신대로 else를 쓰지않았고, dictionary의 value를 가져올 때 get 메소드를 사용하였습니다. 이렇게 사용하니까 코드가 훨씬 간결하게 변한 것 같습니다!